conv/conv.tex

```diff
@@ -0,0 +1,116 @@
+\documentclass[utf8x,hyperref={pdfpagelabels=false}]{beamer}
+
+\usepackage[utf8x]{inputenc}
+\usepackage[OT1]{fontenc}
+\usepackage{graphicx}
+\usepackage{xcolor}
+\usepackage{amsmath}
+
+\usetheme{Malmoe}  % Now it's a beamer presentation with the lisa theme!        
+\usecolortheme{beaver}
+\setbeamertemplate{footline}[page number]
+\setbeamertemplate{navigation symbols}{}
+
+\title{Convolutions for neural networks}
+
+\author{%
+Arnaud Bergeron
+}
+\date{August 10, 2015}
+
+\begin{document}
+
+\begin{frame}[plain]
+ \titlepage
+\end{frame}
+
+\setcounter{page}{1}
+
+\section{Convolution}
+
+% Definitions
+\begin{frame}
+Some vocabulary:
+\begin{description}[feature maps]
+\item[feature map] analogue to layer (the activations, not the weights) for a neural network.
+\item[channel] a single 2D "image" which composes a feature map.
+\item[filter] weight (as in layer weights).
+\item[stack] ???
+\end{description}
+\end{frame}
+
+% Basic operation
+\begin{frame}
+The basic convolution operation (2D, discrete) for a single example
+\begin{overlayarea}{\textwidth}{5.6cm}
+\only<1>{
+\includegraphics[width=\textwidth]{Conv-basic}
+}
+\only<2->{
+\begin{displaymath}
+C_{bo}(i, j) = \sum^{k=0}_{l-1} \sum^{x = 0}_{m-1} \sum^{y = 0}_{n-1} A_{bk}(x + i - \frac{m}{2}, y + j - \frac{n}{2}) F_{ko}(x, y)
+\end{displaymath}
+\begin{itemize}
+\item<3-> $k$ is the input channel.
+\item<4-> $o$ is the output channel.
+\item<5-> $l$ is the number of input channels.
+\item<6-> $(m, n)$ is the size of the filters.
```

Comment:
Where does "m/2" and "n/2" come from? I think this mostly makes sense for "same" convolution, or convolutions where coordinates can be negative, but that is not the case for us.

Maybe you can start with only one input channel and one output channel, which is the regular 2D convolution (in signal processing).